pymongo/lock.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2022-present MongoDB, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import threading
+import weakref
+
+_HAS_REGISTER_AT_FORK = hasattr(os, "register_at_fork")
+
+# References to instances of _create_lock
+_forkable_locks: weakref.WeakSet = weakref.WeakSet()
+
+_insertion_lock = threading.Lock()
```

Comment:
WeakSet.add is implementing using set.add so it is threadsafe and we can remove _insertion_lock and _clients_lock:
https://github.com/python/cpython/blob/3.7/Lib/_weakrefset.py#L81-L84

Reply:
Even asyncio uses WeakSet.add from multiple threads without a lock:
https://github.com/python/cpython/blob/3.7/Lib/asyncio/tasks.py#L860